src/Pool.js

```diff
@@ -207,10 +207,13 @@ Pool.prototype._next = function () {
           .catch(function () {
             // if the worker crashed and terminated, remove it from the pool
             if (worker.terminated) {
-              me._removeWorker(worker);
+              // _removeWorker will call this, but we need it to be removed synchronously
+              me._removeWorkerFromList(worker);
               // If minWorkers set, spin up new workers to replace the crashed ones
               me._ensureMinWorkers();
+              return me._removeWorker(worker);
```

Comment:
Maybe we can change this to something like:

```js
return me._removeWorker(worker).then(function () {
  // If minWorkers set, spin up new workers to replace the crashed ones
  me._ensureMinWorkers();
})
```

So we don't have to clear the worker from the list first? Having to do that feels a bit off, should ideally be only needed to call from within `_removeWorker`.

Reply:
OK, I should probably read all the comments before replying.  But yes.

Reply:
Actually... no.  I think that would have an additional negative affect on performance--you have to wait for termination before spinning up a new one.  I think that just leaving this as-is and removing the call to `_removeWorkerFromList` in `_removeWorker` makes more sense.

Reply:
Or just moving it to before the `Promise`.  Let's just do that...

Reply:
This makes sense 👍